customcommands/tmplextensions.go

```diff
@@ -511,23 +611,51 @@ func tmplDBCount(ctx *templates.Context) interface{} {
 				userID.Int64 = int64(arg)
 				userID.Valid = true
 			case string:
-				keyStr := limitString(arg, 256)
-				key.String = keyStr
-				key.Valid = true
+				patternStr := limitString(arg, 256)
+				pattern.String = patternStr
+				pattern.Valid = true
 			default:
-				return "", errors.New("Invalid Argument Data Type")
+				q, err := queryFromArg(arg)
+				if err!= nil {
+					return "", err
+				}
+				userID = q.UserID
+				pattern = q.Pattern
+				
 			}
 
 		}
 
-		const q = `SELECT count(*) FROM templates_user_database WHERE (guild_id = $1) AND ($2::bigint IS NULL OR user_id = $2) AND ($3::text IS NULL OR key = $3) AND (expires_at IS NULL or expires_at > now())`
+		const q = `SELECT count(*) FROM templates_user_database WHERE (guild_id = $1) AND ($2::bigint IS NULL OR user_id = $2) AND ($3::text IS NULL OR key LIKE $3) AND (expires_at IS NULL or expires_at > now())`
 
 		var count int64
-		err := common.PQ.QueryRow(q, ctx.GS.ID, userID, key).Scan(&count)
+		err := common.PQ.QueryRow(q, ctx.GS.ID, userID, pattern).Scan(&count)
 		return count, err
 	}
 }
 
+func queryFromArg(query interface{}) (*Query, error) {
```

Comment:
honestly seems kinda overkill to json serialize and deserialize, maybe just take in a dict and convert that into a quer?

Reply:
This was being used to create embeds but indeed this is an overkill for a simple structure and I will revise this 👍

Reply:
Should be Okay now @jonas747